### ISSUE TITLE
Data misfit is too high

### ISSUE THREAD

**Issue by AndyMcAliley** [sarafhirsch/EM-CVAE-Graduate_Research#1]
Oh no, data misfit is too hight! Here are some ideas for lowering data misfit:

- Start by tweaking the weights in the loss function:
  - In the final `network = vae.CVAE(...)` in cell 22, the arguments `data_std`, `model_std`, and `beta` determine these weights.
    - Higher beta means the KL divergence term gets weighted more (models will vary less). In theory, beta should be 1, but I found that sometimes it helps if beta is less than 1.
    - Higher data_std increases data misfit (data are treated as being more noisy and therefore aren't fit as closely)
    - Higher model_std increases reconstruction error (encoded models don't match original models as closely)
    - You can think of the overall loss function as $$\text{data misfit} / \text{data std}^2 + \text{model reconstruction error} / \text{model std}^2 + beta * \text{KL divergence}$$
    - So if you increase beta by a factor of four, that would have the same effect as decreasing data_std and model_std by half.
    - See [my thesis](https://repository.mines.edu/bitstream/handle/11124/176515/McAliley_mines_0052E_12209.pdf?sequence=1&isAllowed=y) and [CVAE paper](https://doi.org/10.1190/geo2023-0147.1) for discussion of weighting these terms.
  - All this is to say that you can decrease data_std to try for lower data misfit. However, if data_std is too low, you may see high variance and erratic behavior in the inverted models, and data misfit for models outside the training set may actually go up. You'll have to find the sweet spot through experimentation.
  - For my projects, I had to try several combinations of these to get satisfactory results
- If that doesn't work you could try other approaches, but they'd be more time intensive and experimental.
  - Different model architectures (e.g., CNN or recurrent network for input data, or residual layers in decoder)
  - Different CVAE formulations (https://proceedings.neurips.cc/paper/2015/hash/8d55a249e6baa5c06772297520da2051-Abstract.html, https://doi.org/10.48550/arXiv.1806.02382)
  - Different approaches to training (e.g., changing loss term weights during training)
